tools/protoc-plugin/src/main/java/io/spine/tools/protoc/method/GeneratedMethodGenerator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.protoc.method;
+
+import com.google.common.collect.ImmutableList;
+import io.spine.tools.protoc.CompilerOutput;
+import io.spine.tools.protoc.GenerateMethodsConfig;
+import io.spine.tools.protoc.SpineProtoGenerator;
+import io.spine.tools.protoc.SpineProtocConfig;
+import io.spine.type.MessageType;
+import io.spine.type.Type;
+
+import java.util.Collection;
+
+/**
+ * The {@link SpineProtoGenerator} implementation generating additional message methods.
+ *
+ * <p>The generator produces {@link CompilerOutput compiler output} that fits into the message's
+ * {@link io.spine.tools.protoc.InsertionPoint#CLASS_SCOPE class scope} insertion point.
+ */
+public final class GeneratedMethodGenerator extends SpineProtoGenerator {
```

Comment:
To me `GeneratedMethodGenerator` is way too much. I think it's `MethodGenerator`. Or you could go the other way around and get rid of `Generator` suffix and call it `GeneratedMethod <Something>`.